### PR TITLE
ShoppingCart tests

### PR DESCRIPTION
**Pull request by DZarudnyi**

### CODE REVIEW

**Comment by DZarudnyi** [comment]
image

File: src/main/java/com/example/books/controller/BookController.java [hunk 1]
```diff
@@ -30,22 +30,22 @@
 public class BookController {
     private final BookService bookService;
 
-    @PreAuthorize("hasRole('ROLE_USER')")
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
```
Comment (olekskov) [inline src/main/java/com/example/books/controller/BookController.java:33]: Taking into account that you have only two roles I don't see necessity to put both of them into annotation.
If all roles can access endpoint you can just remove annotation as you security config requires tyo be authenticated anyway
```suggestion
```
Please recheck all similar cases

File: src/main/java/com/example/books/dto/cartitem/CartItemDto.java [hunk 2]
```diff
@@ -2,12 +2,18 @@
...
 
-public record CartItemDto(
-        @NotNull
-        Long bookId,
-        @NotNull
-        @Min(1)
-        int quantity
-) {
+@Data
+public class CartItemDto {
```
Comment (olekskov) [inline src/main/java/com/example/books/dto/cartitem/CartItemDto.java:8]: I don't like approach where part of DTOs are records and part are simple classes.
This is inconsistent approach and each time you use new dto class you should check how to work with it.
Please make all of DTOs either records or classes

File: src/test/resources/database/books/insert-controller-testing-book.sql [hunk 3]
```diff
@@ -1,2 +1,2 @@
 INSERT INTO books(id, title, author, isbn, price, description, cover_image, is_deleted)
-VALUES(123, 'Title1', 'Author1', '1234', 20.00, 'Description', '', false);
\ No newline at end of file
+VALUES(123, 'Title1', 'Author1', '978-0-545-01022-1', 20.00, 'Description', '', false);
```
Comment (olekskov) [inline src/test/resources/database/books/insert-controller-testing-book.sql:2]: Please add empty line in the end of SQL scripts

File: src/test/java/com/example/books/controller/ShoppingCartControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    @Sql(scripts = {
+            "classpath:database/shoppingcart/insert-testing-user.sql",
+            "classpath:database/shoppingcart/insert-testing-shopping-cart.sql",
+            "classpath:database/books/insert-testing-book.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/shoppingcart/remove-testing-shopping-cart.sql",
+            "classpath:database/books/remove-testing-book.sql",
+            "classpath:database/shoppingcart/remove-testing-user.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
```
Comment (olekskov) [inline src/test/java/com/example/books/controller/ShoppingCartControllerTest.java:86]: Please check if it's possible to move majority of scripts to class level to make methods annotations shorter

**Review by olekskov — CHANGES_REQUESTED** [review]

**Comment by DZarudnyi** [comment]
Thank you, that review was very helpful! I actually learned a lot

File: docker-compose.yml [hunk 5]
```diff
@@ -36,4 +36,5 @@ services:
           "spring.datasource.password" : "${MYSQLDB_ROOT_PASSWORD}",
           "spring.datasource.driver-class-name" : "com.mysql.cj.jdbc.Driver"      
         }'
-      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005"
\ No newline at end of file
+#      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005"
+      JAVA_OPTS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (okuzan) [inline docker-compose.yml:40]: remove comment, add empty line at EOF

File: src/test/java/com/example/books/controller/ShoppingCartControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        EqualsBuilder.reflectionEquals(
+                expected,
+                actual
+        );
+    }
+
+    @Test
+    @WithMockUser
+    void deleteCartItem_WithValidId_Ok() throws Exception {
+        mockMvc.perform(delete("/api/cart/cart-items/1"))
```
Comment (okuzan) [inline src/test/java/com/example/books/controller/ShoppingCartControllerTest.java:122]: use your constant

**Review by okuzan — APPROVED** [review]

**Pull request merged**